docs/source/driver/go/snowflake.rst

```diff
@@ -0,0 +1,325 @@
+    the absolute value of such a duration will be used.
+
+``adbc.snowflake.sql.client_option.request_timeout``
+    Specify request retry timeout *excluding* network roundtrip and reading http responses.
+    Value should be formatted as described `here <https://pkg.go.dev/time#ParseDuration>`,
+    such as ``300ms``, ``1.5s`` or ``1m30s``. Even though negative values are accepted,
+    the absolute value of such a duration will be used.
+
+``adbc.snowflake.sql.client_option.jwt_expire_timeout``
+    JWT expiration will occur after this timeout.
+    Value should be formatted as described `here <https://pkg.go.dev/time#ParseDuration>`,
+    such as ``300ms``, ``1.5s`` or ``1m30s``. Even though negative values are accepted,
+    the absolute value of such a duration will be used.
+
+``adbc.snowflake.sql.client_option.client_timeout``
+    Specify timeout for network roundtrip and reading http responses.
+    Value should be formatted as described `here <https://pkg.go.dev/time#ParseDuration>`,
+    such as ``300ms``, ``1.5s`` or ``1m30s``. Even though negative values are accepted,
+    the absolute value of such a duration will be used.
+
+``adbc.snowflake.sql.client_option.app_name``
+    Allows specifying the Application Name to Snowflake for the connection.
+
+``adbc.snowflake.sql.client_option.tls_skip_verify``
+    Disable verification of the server's TLS certificate. Value should be ``true``
+    or ``false``.
+
+``adbc.snowflake.sql.client_option.ocsp_fail_open_mode``
+    Control the fail open mode for OCSP. Default is ``true``. Value should
+    be either ``true`` or ``false``.
+
+``adbc.snowflake.sql.client_option.keep_session_alive``
+    Enable the session to persist even after the connection is closed. Value
+    should be either ``true`` or ``false``.
+
+``adbc.snowflake.sql.client_option.jwt_private_key``
+    Specify the RSA private key which should be used to sign the JWT for
+    authentication. This should be a path to a file containing a PKCS1
+    private key to be read in and parsed. Commonly encoded in PEM blocks
+    of type "RSA PRIVATE KEY".
+
+``adbc.snowflake.sql.client_option.disable_telemetry``
+    The Snowflake driver allows for telemetry information which can be
+    disabled by setting this to ``true``. Value should be either ``true``
+    or ``false``.
+
+``adbc.snowflake.sql.client_option.tracing``
+    Set the logging level
+
+``adbc.snowflake.sql.client_option.cache_mfa_token``
+    When ``true``, the MFA token is cached in the credential manager. Defaults
+    to ``true`` on Windows/OSX, ``false`` on Linux.
+
+``adbc.snowflake.sql.client_option.store_temp_creds``
+    When ``true``, the ID token is cached in the credential manager. Defaults
+    to ``true`` on Windows/OSX, ``false`` on Linux.
+
+
+Metadata
+--------
+
+When calling :cpp:`AdbcConnectionGetTableSchema`, the returned Arrow Schema
+will contain metadata on each field:
```

Comment:
Maybe we should eventually standardize something like Flight SQL's scheme for encoding this sort of info...